Add stage to trigger Hot-Fix pipeline for non-compatible kernels

sireeshapotnuru opened the pull request:
This PR will be merged once Hot-Fix pipeline is fully functional

johnsongeorge-w commented on linux_pipeline/Jenkinsfile_rhel_kernel_validation:
```diff
@@ -172,3 +172,29 @@ stage ("Send Email Report"){
+                VersionLog = readFile("${it}_lis_check.log")
+                if (VersionLog.contains("built-in driver")) {
+                    def Kernelversion = (VersionLog =~ "Running kernel version: (.*).x86_64")[0][1]
+                    TestDistros+="${it}=https://${STORAGE_ACCOUNT}.blob.core.windows.net/kernel/kernel-${Kernelversion}.x86_64.rpm,https://${STORAGE_ACCOUNT}.blob.core.windows.net/kernel/kernel-devel-${Kernelversion}.x86_64.rpm"+"\n"
```
Folder for the storage blob should be used. All the rpm will be under a folder (named 'errata-kernel-version'). Build pipeline will use this folder name to download the rpm from blob.

sireeshapotnuru replied:
Changed the logic to pass "distro"="kernel-version" , since in storage blob a folder will be created with the detected kernel version

johnsongeorge-w commented on linux_pipeline/Jenkinsfile_rhel_kernel_validation:
```diff
@@ -172,3 +172,29 @@ stage ("Send Email Report"){
+            if(TestDistros){
+                TestDistros=TestDistros.trim()
+                echo "TRIGGERING the HOT-FIX pipline for ${TestDistros}"
+                build job: 'Azure/Sandbox/pipeline-lis-hotfix', parameters: [ text(name: 'TARGET_LIST', value: "${TestDistros}")],
```
Remove sandbox and update it accordingly once the pipeline-lis-hotfix is made available.

sireeshapotnuru replied:
Updated the build job path to LIS/pipeline-lis-hotfix

johnsongeorge-w commented:
@sireeshapotnuru : Please address the review comment when the LIS-HOTFIX pipeline is exposed for use. Update the **test result** of integration with lis-hotfix pipeline.

sireeshapotnuru commented:
> @sireeshapotnuru : Please address the review comment when the LIS-HOTFIX pipeline is exposed for use. Update the **test result** of integration with lis-hotfix pipeline.

Addressed the review comments and added a logic to not trigger Hot-Fix pipeline if kernel installation fails

sireeshapotnuru commented:
> Please share the test jobs

please refer below jobs for the success/fail cases of triggering hot-fix from pipeline-lis-validation-redhat-errata-kernel-test20
suceess - job 36
fail - job 40
Note : pipeline-lis-validation-redhat-errata-kernel-test20 is a copy of original pipeline (pipeline-lis-validation-redhat-errata-kernel) which is getting used for testing. Will delete this once PR is merged

sireeshapotnuru commented:
This PR is verified
please refer below jobs for the success/fail cases of triggering hot-fix from pipeline-lis-validation-redhat-errata-kernel-test20
Job 82 - Failed to Trigger Hot-fix pipeline since detected errata kernel installation is failed
Job 84 - Triggered Hot-fix pipeline

iamshital commented on linux_pipeline/Jenkinsfile_rhel_kernel_validation:
```diff
@@ -173,3 +173,37 @@ stage ("Send Email Report"){
+            return
+        }
+        def TestDistros = ""
+        def TriggerHotfix = true
```
TriggerHotfix should be parameter of this pipeline with default value = 'true'

iamshital commented on linux_pipeline/Jenkinsfile_rhel_kernel_validation:
```diff
@@ -173,3 +173,37 @@ stage ("Send Email Report"){
+        }
+        def TestDistros = ""
+        def TriggerHotfix = true
+        dir ("trigger-hotfix-${env.BUILD_NUMBER}-${env.BRANCH_NAME}") {
```
And if the TriggerHotfix is give, then proceed for further tests.

iamshital commented on linux_pipeline/Jenkinsfile_rhel_kernel_validation:
```diff
@@ -173,3 +173,37 @@ stage ("Send Email Report"){
+stage ("Trigger Hotfix"){
+    node ("jenkins-meta-slave") {
+        if (!sendMail) {
+            return
```
What is the use of this?

sireeshapotnuru replied:
This helps to return from each stage when there is no new errata kernel detected
